apps/test/unit/templates/courseRollupPages/RollupLessonEntrySectionTest.jsx

```diff
@@ -0,0 +1,136 @@
+    defaultProps = {
+      objectToRollUp: 'Resources',
+      lesson: courseData.units[0].lessons[0]
+    };
+  });
+
+  it('renders list of resources when there are resources', () => {
+    const wrapper = mount(<RollupLessonEntrySection {...defaultProps} />);
+
+    expect(wrapper.find('ResourceList').length).to.equal(2);
+  });
+
+  it('renders no resources message when no resources', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        lesson={courseData.units[1].lessons[0]}
+      />
+    );
+
+    expect(wrapper.containsMatchingElement(<p>{i18n.rollupNoResources()}</p>))
+      .to.be.true;
+  });
+
+  it('renders list of prep when there is prep', () => {
+    const wrapper = mount(
+      <Provider store={getStore()}>
+        <RollupLessonEntrySection {...defaultProps} objectToRollUp={'Prep'} />
+      </Provider>
+    );
+
+    expect(
+      wrapper.containsMatchingElement(
+        <EnhancedSafeMarkdown
+          markdown={defaultProps.lesson.preparation}
+          expandableImages
+        />
+      )
+    ).to.be.true;
+  });
+
+  it('renders no prep message when no prep', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        objectToRollUp={'Prep'}
+        lesson={courseData.units[1].lessons[0]}
+      />
+    );
+
+    expect(wrapper.containsMatchingElement(<p>{i18n.rollupNoPrep()}</p>)).to.be
+      .true;
+  });
+
+  it('renders list of vocab when there is vocab', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        objectToRollUp={'Vocabulary'}
+      />
+    );
+
+    expect(
+      wrapper.containsMatchingElement(
+        <li key={defaultProps.lesson.vocabularies[0].key}>
+          <InlineMarkdown
+            markdown={`**${defaultProps.lesson.vocabularies[0].word}** - ${
+              defaultProps.lesson.vocabularies[0].definition
+            }`}
+          />
+        </li>
+      )
+    ).to.be.true;
+  });
+
+  it('renders no vocab message when no vocab', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection
+        {...defaultProps}
+        objectToRollUp={'Vocabulary'}
+        lesson={courseData.units[1].lessons[0]}
+      />
+    );
+
+    expect(wrapper.containsMatchingElement(<p>{i18n.rollupNoVocab()}</p>)).to.be
+      .true;
+  });
+
+  it('renders list of code when there is code', () => {
+    const wrapper = mount(
+      <RollupLessonEntrySection {...defaultProps} objectToRollUp={'Code'} />
+    );
+
+    expect(
+      wrapper.containsMatchingElement(
+        <li key={defaultProps.lesson.programmingExpressions[0].name}>
+          <a
+            href={studio(defaultProps.lesson.programmingExpressions[0].link)}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {defaultProps.lesson.programmingExpressions[0].name}
+          </a>
+        </li>
+      )
+    ).to.be.true;
```

Comment:
too specific. I would suggest checking that the name appears in the text, and that there is an element matching `<a href={studio(defaultProps.lesson.programmingExpressions[0].link)}>`. otherwise you are testing the internals of the component that renders the programming expressions.

Reply:
sorry to harp on this, but https://github.com/code-dot-org/code-dot-org/pull/39716 is actually a great example of a subsequent change which should not break this test.

Reply:
So instead of checking for the StyledCodeBlock I should check for the block name and link?

Reply:
yes, that's what I'd recommend.

Reply:
the practical consideration is to make the tests less brittle. the theoretical consideration is that unit tests should not reach in and test the details of what other components are doing. there is a big overlap between these two principles. However, since it seems like a safe bet that the StyledCodeBlock or whatever replaces it will eventually render the block name and link, what I'm suggesting seems to meet the practical goal best.

checking for exact params passed to StyledCodeBlock would follow the "don't reach inside other components" rule  the most strictly, and could also be ok. I guess what I am suggesting is a balance between the practical and theoretical principles since it seems least likely to break. in theory that might make this a mini integration test of sorts, but I digress...